src/Nest/Modules/SnapshotAndRestore/Repositories/S3Repository.cs

```diff
@@ -7,131 +7,271 @@ public interface IS3Repository : IRepository<IS3RepositorySettings> { }
-		/// <param name="compress"></param>
-		public S3RepositorySettingsDescriptor Compress(bool compress = true) => Assign(a => a.Compress = compress);
-
-		/// <summary>
-		/// Throttles the number of streams (per node) preforming snapshot operation. Defaults to 5
-		/// </summary>
-		/// <param name="concurrentStreams"></param>
-		public S3RepositorySettingsDescriptor ConcurrentStreams(int concurrentStreams) => Assign(a => a.ConcurrentStreams = concurrentStreams);
+		/// <inheritdoc cref="IS3RepositorySettings.ConcurrentStreams"/>
+		public S3RepositorySettingsDescriptor ConcurrentStreams(int? concurrentStreams) => Assign(a => a.ConcurrentStreams = concurrentStreams);
+#pragma warning restore 618
 
-		/// <summary>
-		///  Big files can be broken down into chunks during snapshotting if needed.
-		/// The chunk size can be specified in bytes or by using size value notation,
-		/// i.e. 1g, 10m, 5k. Defaults to 100m.
-		/// </summary>
-		/// <param name="chunkSize"></param>
+		/// <inheritdoc cref="IS3RepositorySettings.ChunkSize"/>
 		public S3RepositorySettingsDescriptor ChunkSize(string chunkSize) => Assign(a => a.ChunkSize = chunkSize);
+
+		/// <inheritdoc cref="IS3RepositorySettings.Compress"/>
+		public S3RepositorySettingsDescriptor Compress(bool? compress = true) => Assign(a => a.Compress = compress);
+
+		/// <inheritdoc cref="IS3RepositorySettings.ServerSideEncryption"/>
+		public S3RepositorySettingsDescriptor ServerSideEncryption(bool? serverSideEncryption = true) =>
+			Assign(a => a.ServerSideEncryption = serverSideEncryption);
+
+		/// <inheritdoc cref="IS3RepositorySettings.BufferSize"/>
+		public S3RepositorySettingsDescriptor BufferSize(string bufferSize) => Assign(a => a.BufferSize = bufferSize);
+
+#pragma warning disable 618
+		/// <inheritdoc cref="IS3RepositorySettings.MaximumRetries"/>
+		public S3RepositorySettingsDescriptor MaximumRetries(int maximumRetries) => Assign(a => a.MaximumRetries = maximumRetries);
+
+		/// <inheritdoc cref="IS3RepositorySettings.UseThrottleRetries"/>
+		public S3RepositorySettingsDescriptor UseThrottleRetries(bool? useThrottleRetries = true) => Assign(a => a.UseThrottleRetries = useThrottleRetries);
+
+		/// <inheritdoc cref="IS3RepositorySettings.ReadOnly"/>
+		public S3RepositorySettingsDescriptor ReadOnly(bool? @readonly = true) => Assign(a => a.ReadOnly = @readonly);
+#pragma warning restore 618
+
+		/// <inheritdoc cref="IS3RepositorySettings.CannedAcl"/>
+		public S3RepositorySettingsDescriptor CannedAcl(string cannedAcl) => Assign(a => a.CannedAcl = cannedAcl);
+
+		/// <inheritdoc cref="IS3RepositorySettings.StorageClass"/>
+		public S3RepositorySettingsDescriptor StorageClass(string storageClass) => Assign(a => a.StorageClass = storageClass);
+
+#pragma warning disable 618
+		/// <inheritdoc cref="IS3RepositorySettings.PathStyleAccess"/>
+		public S3RepositorySettingsDescriptor PathStyleAccess(bool? pathStyleAccess = true) => Assign(a => a.PathStyleAccess = pathStyleAccess);
```

Comment:
These also need the `[Obsolete]` attribute? Then can remove the pragma directives

Reply:
:+1: